components/emoji/render_emoji.tsx

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React from 'react';
+import {useSelector} from 'react-redux';
+
+import {getEmojiImageUrl} from 'mattermost-redux/utils/emoji_utils';
+
+import {getEmojiMap} from 'selectors/emojis';
+import {GlobalState} from 'types/store';
+interface ComponentProps {
+    emojiName: string;
+    size?: number;
+    emojiStyle?: React.CSSProperties;
+    onClick?: () => void;
+}
+
+const RenderEmoji = ({emojiName, emojiStyle, size, onClick}: ComponentProps) => {
```

Comment:
this is where the component should probably wrapped in `React.memo( ... )` since a shallow compare for the CSS properties would prevent a re-render when they are equal